athena-hbase/src/main/java/com/amazonaws/athena/connectors/hbase/integ/HbaseTableUtils.java

```diff
@@ -88,6 +98,41 @@ private Configuration getHbaseConfiguration(String connectionStr)
         configuration.set("hbase.client.pause", "500");
         configuration.set("zookeeper.recovery.retry", "2");
 
+        java.util.Map<String, String> configOptions = System.getenv();
+        boolean kerberosAuthEnabled = configOptions.get(KERBEROS_AUTH_ENABLED) != null && "true".equalsIgnoreCase(configOptions.get(KERBEROS_AUTH_ENABLED));
+        logger.info("Kerberos Authentication Enabled: " + kerberosAuthEnabled);
+        if (kerberosAuthEnabled) {
+            String keytabLocation = null;
+            configuration.set("hbase.rpc.protection", configOptions.get(HBASE_RPC_PROTECTION));
+            logger.info("hbase.rpc.protection: " + configuration.get("hbase.rpc.protection"));
+            String s3uri = configOptions.get(KERBEROS_CONFIG_FILES_S3_REFERENCE);
+            if (StringUtils.isNotBlank(s3uri)) {
+                try {
+                    Path tempDir = copyConfigFilesFromS3ToTempFolder(configOptions);
+                    logger.debug("tempDir: " + tempDir);
+                    keytabLocation = tempDir + File.separator + "hbase.keytab";
+                    System.setProperty("java.security.krb5.conf", tempDir + File.separator + "krb5.conf");
+                    logger.debug("krb5.conf location: " + tempDir + File.separator + "krb5.conf");
+                }
+                catch (Exception e) {
+                    logger.error("Exception while copying config files from S3 to temp folder: " + e.getMessage());
+                    throw new RuntimeException(e);
+                }
+            }
+            logger.debug("keytabLocation: " + keytabLocation);
+
+            UserGroupInformation.setConfiguration(configuration);
+            try {
+                String principalName = configOptions.get(PRINCIPAL_NAME);
+                UserGroupInformation.loginUserFromKeytab(principalName, keytabLocation);
+            }
+            catch (IOException ex) {
```

Comment:
++++

Reply:
Hi @AbdulR3hman, updated in latest commit.

Reply:
see [`public RuntimeException(String message,Throwable cause)`](https://docs.oracle.com/javase/8/docs/api/java/lang/RuntimeException.html#RuntimeException-java.lang.String-java.lang.Throwable-)

Reply:
Hi @AbdulR3hman, Updated code as per your suggestion.